src/fides/api/service/privacy_request/request_service.py

```diff
@@ -320,3 +323,119 @@ def remove_saved_dsr_data(self: DatabaseTask) -> None:
+    2. Fetches all in-progress privacy requests from the database.
+    3. Checks each privacy request to determine if its tasks are still active.
+    4. Requeues the privacy request if any of its tasks are found to be interrupted.
+    """
+
+    with self.get_new_session() as db:
+        logger.debug("Starting check for interrupted tasks to requeue")
+
+        # Get all task IDs currently in the queue
+        redis_conn: FidesopsRedis = get_cache()
+        queued_tasks = redis_conn.lrange("fides.dsr", 0, -1)
+        queued_tasks_ids = [json.loads(task)["headers"]["id"] for task in queued_tasks]
+
+        # Get all in-progress privacy requests
+        in_progress_requests = (
+            db.query(PrivacyRequest)
+            .filter(
+                PrivacyRequest.status.in_(
+                    [PrivacyRequestStatus.in_processing, PrivacyRequestStatus.approved]
+                )
+            )
+            .filter(PrivacyRequest.deleted_at.is_(None))
+            .order_by(PrivacyRequest.created_at)
+        )
+
+        logger.debug(f"Found {in_progress_requests.count()} privacy requests to check")
+
+        # Check each privacy request
+        for privacy_request in in_progress_requests:
+            should_requeue = False
+            logger.debug(f"Checking tasks for privacy request {privacy_request.id}")
+            task_id = privacy_request.get_cached_task_id()
+
+            # If the task ID is not cached, we can't check if it's running
+            if not task_id:
+                continue
+
+            # Check if the main privacy request task is active
+            if task_id not in queued_tasks_ids and not celery_tasks_in_flight(
+                [task_id]
+            ):
+                if privacy_request.request_tasks.count() == 0:
+                    logger.warning(
+                        f"The task for privacy request {privacy_request.id} was terminated before it could schedule any request tasks, requeueing privacy request"
+                    )
+                    should_requeue = True
+
+                # Check each individual request task
+                for request_task in privacy_request.request_tasks:
+                    if request_task.status in [
+                        ExecutionLogStatus.in_processing,
+                        ExecutionLogStatus.pending,
+                    ]:
+                        subtask_id = request_task.get_cached_task_id()
+
+                        # If the task ID is not cached, we can't check if it's running
+                        if not subtask_id:
+                            continue
+
+                        if (
+                            subtask_id not in queued_tasks_ids
+                            and not celery_tasks_in_flight([subtask_id])
+                        ):
+                            logger.warning(
+                                f"Request task {request_task.id} is not in the queue or running"
```

Comment:
As long as I'm reading this correctly, this requeues the privacy request, right?
If so, we should add `, requeueing privacy request` like we have above.

```suggestion
                                f"Request task {request_task.id} is not in the queue or running, requeueing privacy request"
```